BUILDBOT.md

```diff
@@ -0,0 +1,134 @@
+compatibility see the
+[documentation](https://learn.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility)
+
+| Tag Name  | Automated | Description |
+|---|:---:|---|
+| 2022 | :white_check_mark: | A Windows 2022 server container |
+| windows-2022 | :x: | A Windows container, used for Layout Tests |
+| 1809 | :x: | A Windows 2019 server container |
+
+Visit [Docker Hub](https://hub.docker.com/r/webkitdev/buildbot-worker/tags) to
+see when the last build happened for the image's tags.
+
+## Configuring the Buildbot worker
+The `webkitdev/buildbot-worker` image uses environment variables for
+configuration. When invoking `docker run` environment variables are specified on
+the command line individually or through a file. See the
+[documentation](https://docs.docker.com/engine/reference/commandline/container_run/#env)
+for more details.
+
+| Environment Variable  | Required | Description |
+|---|:---:|---|
+| BUILD_HOST_NAME | :white_check_mark: | Host of the buildbot instance, either `build.webkit.org` or `ews-build.webkit.org` |
+| BUILD_HOST_PORT | :x: | Port of the buildbot instance, defaults to 17000 |
+| BUILD_WORKER_NAME | :white_check_mark: | Name for the worker, provided by infrastructure team  |
+| BUILD_WORKER_PASSWORD | :white_check_mark: | Password for the worker, provided by infrastructure team |
+| BUILD_WORKER_KEEPALIVE | :x: | Time in seconds that messages should be sent by the worker to the server, defaults to 240 |
+| ADMIN_NAME | :x:| Contact name for the admin of the worker | 
+| ADMIN_EMAIL | :x: | Contact e-mail address for the admin of the worker |
+| HOST_DESCRIPTION |:x: | A description of the host running the worker |
+| COMPILER | :x: | Set this to `Clang` to use `clang-cl`, defaults to `cl` |
+
+The preferred method of setup uses an environment file. An example file to fill
+out with all the options present looks like this. Save the text file locally
+with the name corresponding to worker name; for this example `build.env` is
+used.
+
+```txt
+ADMIN_EMAIL=<email>
+ADMIN_NAME=<admin>
+BUILD_HOST_NAME=<host>
+BUILD_WORKER_NAME=<name>
+BUILD_WORKER_PASSWORD=<password>
+HOST_DESCRIPTION=<description>
+COMPILER=Clang
+```
+
+## Running the Buildbot worker
+On a Windows Server 2022 machine run the following in a powershell session. On
+Windows Server 2022 the Docker container runs in process isolation mode which
+gives it access to all the resources of the host.
+
+```powershell
+docker run --name build --env-file build.env --detach --restart always webkitdev/buildbot-worker:2022
+```
+
+On a Windows 11 machine run the following in a powershell session. On Windows 11
+the Docker container runs in Hyper-V isolation mode so the amount of resources
+need to be specified with `cpu-count` and `memory`. Replace `X` with the number
+of processors to use and `Y` with the amount of memory in GBs to reserve for the
+container.
```

Comment:
We should mention any minimums that will fail the bot. Also, do we need to worry about the storage space for this case?